main.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Cache interface {
+	Set(key string, value interface{}, ttl time.Duration)
+	Get(key string) (interface{}, bool)
+	Delete(key string)
+}
+
+type Self struct{}
```

Comment:
What was the purpose of having this type here? Also, name `Self` is not the best one in my opinion. Name of type should represent its meaning, e.g. `http.Client`, `net.IP` etc.